src/calculator/index.ts

```diff
@@ -186,12 +186,22 @@ export default class Calculator {
       round.metadata?.quadraticFundingConfig?.sybilDefense ??
       false;
 
+    // 1. convert threshold amount to 6 decimals
+    // 2. truncate rest of decimals to bigint
+    // 3. convert decimals to token decimals
+    // 4. remove initial 6 decimals
     const minimumAmount =
       this.minimumAmount ??
-      BigInt(
-        round.metadata?.quadraticFundingConfig?.minDonationThresholdAmount ?? 0
+      (BigInt(
+        Math.trunc(
+          Number(
```

Comment:
This nested bit can be a bit difficult to read and I would not be opposed to breaking this into its own function. Something like: 
```
function calculateMinimumAmount(
  minDonationThresholdAmount: number | undefined,
  matchTokenDecimals: bigint
): bigint {
  if (minDonationThresholdAmount === undefined) {
    return 0n;
  }

  const sixDecimalMultiplier = Math.pow(10, 6);
  const truncatedAmount = Math.trunc(minDonationThresholdAmount * sixDecimalMultiplier);
  const bigIntTruncatedAmount = BigInt(truncatedAmount);

  const decimalConversionFactor = 10n ** matchTokenDecimals / 10n ** 6n;
  const minimumAmount = bigIntTruncatedAmount * decimalConversionFactor;

  return minimumAmount;
}

const minDonationThresholdAmount = round.metadata?.quadraticFundingConfig?.minDonationThresholdAmount;
const minimumAmount = this.minimumAmount ?? calculateMinimumAmount(minDonationThresholdAmount, matchTokenDecimals) ?? 0n;

let matchingCapAmount = this.matchingCapAmount;
```
but thats just my opinion! lgtm :D

Reply:
Merging now so we can test the E2E flow and will refactor Later™